launchpod/angular-launchpod/backend/src/main/java/com/google/launchpod/servlets/TTSServlet.java

```diff
@@ -0,0 +1,238 @@
+
+        Query query = new Query(LoginStatus.USER_FEED_KEY).addSort(LoginStatus.TIMESTAMP_KEY, SortDirection.DESCENDING);
+
+        PreparedQuery results = datastore.prepare(query);
+
+        ArrayList<UserFeed> userFeeds = new ArrayList<UserFeed>();
+        for (Entity entity : results.asIterable()) {
+            if (userEmail.equals(entity.getProperty(EMAIL).toString())) {
+                String userFeedTitle = (String) entity.getProperty(LoginStatus.TITLE_KEY);
+                String userFeedName = (String) entity.getProperty(LoginStatus.NAME_KEY);
+                String userFeedDescription = (String) entity.getProperty(LoginStatus.DESCRIPTION_KEY);
+                String userFeedLanguage = (String) entity.getProperty(LoginStatus.LANGUAGE_KEY);
+                String userFeedEmail = (String) entity.getProperty(LoginStatus.EMAIL_KEY);
+                long userFeedTimestamp = (long) entity.getProperty(LoginStatus.TIMESTAMP_KEY);
+                Date date = new Date(userFeedTimestamp);
+                SimpleDateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy  HH:mm:ss Z", Locale.getDefault());
+                String postTime = dateFormat.format(date);
+                Key key = entity.getKey();
+
+                String urlID = KeyFactory.keyToString(entity.getKey()); // the key string associated with the entity,
+                                                                        // not the
+                                                                        // numeric ID.
+                String rssLink = BASE_URL + urlID;
+
+                userFeeds.add(new UserFeed(userFeedTitle, userFeedName, rssLink, userFeedDescription, userFeedEmail,
+                        postTime, urlID, userFeedLanguage));
+            }
+        }
+
+        res.setContentType("application/json");
+        res.getWriter().println(GSON.toJson(userFeeds));
+    }
+
+    /**
+     * doGet obtains an ID from the link to search for specific blob in cloud
+     * storage. After obtaining the blob, we get the bytes then stream that to the
+     * client using the server outputstream
+     * 
+     * @throws IOException when unable to run doGet method
+     */
+    public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException {
+        res.setContentType("audio/mpeg");
+        String id = req.getParameter(ID);
+
+        if (Strings.isNullOrEmpty(id)) {
+            throw new IllegalArgumentException("Invalid URL. Please try again");
+        }
+
+        // Search for id in cloud storage from parameter ID
+        Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+        Blob desiredFeedBlob = storage.get(BUCKET_NAME, id);
+        byte[] blobBytes = desiredFeedBlob.getContent(BlobSourceOption.generationMatch());
+
+        // Write bytes to servlet output stream
+        res.getOutputStream().write(blobBytes);
+        res.getWriter().print(blobBytes.toString());
```

Comment:
do we need both? looks like they're doing the same thing

seems like it's recommended to use res.getOutputStream().write(blobBytes) for binary data: https://stackoverflow.com/questions/26010650/servlet-response-write-vs-print-which-is-better